misc: Add release notes for version 23.1

Harshil2107 opened the pull request:
- Updated release notes for gem5 version 23.1 with workload resource specialization, suites and new env variables

Change-Id: I9e6097b37432b862b3a212440a79c3c0da836ec9

Harshil2107 committed:
stdlib: Updated release notes

- Updated release notes for gem5 version 23.1
with workload resource specialization, suites and new env variables

Change-Id: I9e6097b37432b862b3a212440a79c3c0da836ec9

powerjg commented on RELEASE-NOTES.md:
```diff
@@ -1,3 +1,20 @@
+# Version 23.1
+## Significant API and user-facing changes
+### `WorkloadResource` added to resource specialization
+- `Workload` and `AbstractWorkload` classes have been deprecated. They have been converted to be wrappers of `obtain_resource` and `WorkloadResource` class respectively.
```
How does one fix their code if they are using the old API?

Harshil2107 commented:
Note: Anyone may add a commit to this PR to add their change. Please do not do rebases. This will be submitted as a  Squash merge.

giactra commented:
Is this just for stdlib? Or for any update to the RELEASE file?

Harshil2107 commented:
I think this is for any updates we want to have in RELEASE-NOTES.md file.

abmerop commented:
> Note: Anyone may add a commit to this PR to add their change. Please do not do rebases. This will be submitted as a Squash merge.

I'm sure it's possible. But... how? I don't know how to just 'add a commit' I added your repo as a remote and tried to push. Are there notes somewhere? I want to add release notes for gpufs.

Harshil2107 commented:
> I'm sure it's possible. But... how? I don't know how to just 'add a commit' I added your repo as a remote and tried to push. Are there notes somewhere? I want to add release notes for gpufs.

This is the github documentation on pushing to a PR from a fork: https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/committing-changes-to-a-pull-request-branch-created-from-a-fork

I have also selected the option to allow edits by maintainers so others should be able to push to this PR.

I also use the github PR and issue extension on VSCode which allows users to checkout PRs and push to them. The extension id is: GitHub.vscode-pull-request-github

![image](https://github.com/gem5/gem5/assets/91860903/98895c12-d8ae-436c-8dd4-4ac85f5f052d)

You can click the arrow near the description to checkout the PR.

BobbyRBruce commented:
I personally like using the GitHub command line utility: https://cli.github.com. It allows you to checkout a pull request. From there it's just adding a commit to the checkout branch and `git push` from there.

abmerop committed:
arch-vega,gpu-compute,dev-amdgpu: Update release notes

Update release notes for GPU full-system specific changes for v23.1.

Change-Id: I1827dcda45d8fdd6d0704e27f746df20850539b4

mattsinc commented:
@abmerop : just confirming I need to add the other GPU/VIPER Ruby updates besides GPUFS stuff to the release notes?

abmerop commented:
> @abmerop : just confirming I need to add the other GPU/VIPER Ruby updates besides GPUFS stuff to the release notes?

Yes you will need to do that.  I have only added release notes from my own PRs.

Harshil2107 committed:
misc: Updated release notes

- Added known issues
- Added building gem5 with kconfig

Change-Id: I5900ce9f632c1c975456e4790e43948e46ed03e6

Harshil2107 committed:
misc: updated file formating

Change-Id: I6f75ec9b1caf2114e28d4f587728be71e66d6488

powerjg committed:
misc: Update release notes

Change-Id: I6f269df9fd66740e93a43aa4021b0d944406232e
Signed-off-by: Jason Lowe-Power <jason@lowepower.com>

Harshil2107 committed:
misc: Update release notes

Change-Id: I7f5c4e3d3902790fb59e02d519212c3261d8807e

powerjg commented on RELEASE-NOTES.md:
```diff
@@ -34,6 +34,8 @@ Suites is a new category of resource being introduced in gem5. Documentation of
 - Added support for clang 15 and clang 16
 - gem5 no longer supports building on Ubuntu 18.04
 - GCC 7, GCC 9, and clang 6 are no longer supported
+- The stats `board.memory.mem_ctrl.dram.bytesRead` and `board.memory.mem_ctrl.dram.bytesWritten` are changed to `board.memory.mem_ctrl.dram.dramBytesRead` and `board.memory.mem_ctrl.dram.dramBytesWritten` respectively.
+- The stats `board.memory.mem_ctrl.nvm.bytesRead` and `board.memory.mem_ctrl.nvm.bytesWritten` are changed to `board.memory.mem_ctrl.nvm.nvmBytesRead` and `board.memory.mem_ctrl.nvm.nvmBytesWritten` respectively.
```
@Harshil2107 

```suggestion
- Two `DRAMInterface` stats have changed names (`bytesRead` and `bytesWritten`). For instance, `board.memory.mem_ctrl.dram.bytesRead` and `board.memory.mem_ctrl.dram.bytesWritten`. These are changed to `dramBytesRead` and `dramBytesWritten` so they don't collide with the stat with the same name in `AbstractMemory`.
- The stats for `NVMInterface` (`bytesRead` and `bytesWritten`) have been change to `nvmBytesRead` and `nvmBytesWritten` as well.
```

giactra committed:
misc: Update release notes with ArmISA section

The commit is also adding a TraceCPU entry in the generic
improvement category

Change-Id: Ie1fd0bca1a91fd5152d2a485e2e803a389d5f66b
Signed-off-by: Giacomo Travaglini <giacomo.travaglini@arm.com>

powerjg commented:
Thanks @giactra and @abmerop! @mattsinc please add any updates that you see fit. You can either directly push to this PR or you can put them in a comment and Harshil will copy them in. We're planning to make the release at the end of the day today, if all goes according to plan :)

abmerop commented:
This looks good from my end. @mattsinc just making sure, any GPU-SE notes to add?  

Otherwise, congrats folks on the release!

mattsinc commented:
Sorry, I will work on this today.  Trying to release this during final exams has not left me much time to help, sorry.

Harshil2107 committed:
misc: Update release notes

Change-Id: I53c4d98c39729ee9d4a2e4f62b08ef4530ba34a8

Harshil2107 committed:
misc: Update release notes

Change-Id: I53c4d98c39729ee9d4a2e4f62b08ef4530ba34a8

mattsinc committed:
misc: Update release notes for version 23.1

Add more GPU stuff

Change-Id: I21415edde6107f8433d74607b9a3f0211416b8cd

mattsinc committed:
misc: Add release notes for version 23.1

Change-Id: I21415edde6107f8433d74607b9a3f0211416b8cd

mattsinc committed:
Merge branch 'release-notes-update' of https://github.com/Harshil2107/gem5 into release-notes-update

Change-Id: I74c9a29a179bc7488499350285f1e68ceadb6e10

mattsinc committed:
Merge branch 'release-notes-update' of https://github.com/Harshil2107/gem5 into release-notes-update

Change-Id: Iebf519c2529d573a0b7c0ac02380016f18fd317c

mattsinc commented:
I am not 100% sure my updates worked properly when I tried to push them to Harshil's branch, let me know if something is messed up

rogerchang23424 commented on RELEASE-NOTES.md:
```diff
@@ -1,3 +1,130 @@
+
+Most gem5 builds (e.g., `build/X86/gem5.opt`) are backwards compatible and require no changes to your current workflows.
+All of the default builds in `build_opts` are unchanged and still available.
+However, if you want to specialize your build, you now have to use `scons <kconfig command>`.
```
Sorry for lately reply, I would like to add following description:

> However, if you want to specialize your build, you now have to use `scons <kconfig command>`.

To

> However, if you want to specialize your build. For example, use customized ruby protocol. The command `scons PROTOCOL=<PROTOCAL_NAME> build/ALL/gem5.opt` will not work anymore. you now have to use `scons <kconfig command>` to update the ruby protocol as example. The double dash options (--without-tcmalloc, --with-asan and so on) are still continue to work as normal.

rogerchang23424 commented on RELEASE-NOTES.md:
```diff
@@ -1,3 +1,130 @@
+
+### The gem5 build can is now configured with `kconfig`
+
+Most gem5 builds (e.g., `build/X86/gem5.opt`) are backwards compatible and require no changes to your current workflows.
```
Most gem5 builds without customized options (excluding double dash options) (e.g. ...

Harshil2107 replied:
Just to confirm do you want the sentence to change to "Most gem5 builds without customized options (excluding double dash options) (e.g. , `build/X86/gem5.opt`) are backwards compatible and require no changes to your current workflows."?

rogerchang23424 replied:
Yes

rogerchang23424 commented on RELEASE-NOTES.md:
```diff
@@ -1,3 +1,130 @@
+  - Widening/narrowing instructions are *not* implemented
+  - The model for executing memory instructions is not very high performance
+  - The statistics are not correct for counting vector instruction execution
+
```
> Add new RISC-V Zcb instructions
https://github.com/gem5/gem5/pull/399

Harshil2107 committed:
misc: Added Roger's suggestions

Change-Id: I93ee69fccb4d9477559a46fe34dfc86547ca6f22
